libraries/botbuilder/src/recognizer.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+import { Middleware, Promiseable } from './middleware';
+import { RecognizerResult } from './recognizerResult';
+import { Intent } from './intentRecognizer';
+
+
+/**
+ * Middleware that's the base class for all intent recognizers.
+ * 
+ * __Extends BotContext:__
+ * * context.topIntent - The top recognized `Intent` for the users utterance.
+ */
+export class Recognizer implements Middleware {
```

Comment:
>Recognizer [](start = 13, length = 10)

Why did you have to add a new Recognizer class? Is it because the result type changed?

Reply:
@Stevenic yes, eventually all recognizers should implement this instead of IntentRecognizer and we should delete the old IntentRecognizer, I just didn't want to change the old one till we agree on the new schema and have it working for one recognizer.